// practing-ui@0.0.1/dist/practing-ui.mjs
function cn(e, t) {
  const n = /* @__PURE__ */ Object.create(null), s = e.split(",");
  for (let r = 0; r < s.length; r++)
    n[s[r]] = !0;
  return t ? (r) => !!n[r.toLowerCase()] : (r) => !!n[r];
}
const P = process.env.NODE_ENV !== "production" ? Object.freeze({}) : {}, ln = process.env.NODE_ENV !== "production" ? Object.freeze([]) : [], Ue = () => {
}, un = /^on[^a-z]/, an = (e) => un.test(e), x = Object.assign, fn = (e, t) => {
  const n = e.indexOf(t);
  n > -1 && e.splice(n, 1);
}, pn = Object.prototype.hasOwnProperty, m = (e, t) => pn.call(e, t), h = Array.isArray, Q = (e) => Ve(e) === "[object Map]", dn = (e) => Ve(e) === "[object Set]", E = (e) => typeof e == "function", $ = (e) => typeof e == "string", Ge = (e) => typeof e == "symbol", O = (e) => e !== null && typeof e == "object", hn = (e) => O(e) && E(e.then) && E(e.catch), _n = Object.prototype.toString, Ve = (e) => _n.call(e), bt = (e) => Ve(e).slice(8, -1), gn = (e) => Ve(e) === "[object Object]", Je = (e) => $(e) && e !== "NaN" && e[0] !== "-" && "" + parseInt(e, 10) === e, mn = (e) => {
  const t = /* @__PURE__ */ Object.create(null);
  return (n) => t[n] || (t[n] = e(n));
}, En = mn(
  (e) => e.charAt(0).toUpperCase() + e.slice(1)
), se = (e, t) => !Object.is(e, t), wn = (e, t, n) => {
  Object.defineProperty(e, t, {
    configurable: !0,
    enumerable: !1,
    value: n
  });
};
let nt;
const Me = () => nt || (nt = typeof globalThis < "u" ? globalThis : typeof self < "u" ? self : typeof window < "u" ? window : typeof global < "u" ? global : {});
function xe(e) {
  if (h(e)) {
    const t = {};
    for (let n = 0; n < e.length; n++) {
      const s = e[n], r = $(s) ? On(s) : xe(s);
      if (r)
        for (const o in r)
          t[o] = r[o];
    }
    return t;
  } else {
    if ($(e))
      return e;
    if (O(e))
      return e;
  }
}
const Nn = /;(?![^(]*\))/g, bn = /:([^]+)/, vn = new RegExp("\\/\\*.*?\\*\\/", "gs");
function On(e) {
  const t = {};
  return e.replace(vn, "").split(Nn).forEach((n) => {
    if (n) {
      const s = n.split(bn);
      s.length > 1 && (t[s[0].trim()] = s[1].trim());
    }
  }), t;
}
function Ie(e) {
  let t = "";
  if ($(e))
    t = e;
  else if (h(e))
    for (let n = 0; n < e.length; n++) {
      const s = Ie(e[n]);
      s && (t += s + " ");
    }
  else if (O(e))
    for (const n in e)
      e[n] && (t += n + " ");
  return t.trim();
}
function rt(e, ...t) {
  console.warn(`[Vue warn] ${e}`, ...t);
}
let vt;
function Sn(e, t = vt) {
  t && t.active && t.effects.push(e);
}
function yn() {
  return vt;
}
const oe = (e) => {
  const t = new Set(e);
  return t.w = 0, t.n = 0, t;
}, Ot = (e) => (e.w & j) > 0, St = (e) => (e.n & j) > 0, Vn = ({ deps: e }) => {
  if (e.length)
    for (let t = 0; t < e.length; t++)
      e[t].w |= j;
}, xn = (e) => {
  const { deps: t } = e;
  if (t.length) {
    let n = 0;
    for (let s = 0; s < t.length; s++) {
      const r = t[s];
      Ot(r) && !St(r) ? r.delete(e) : t[n++] = r, r.w &= ~j, r.n &= ~j;
    }
    t.length = n;
  }
}, Fe = /* @__PURE__ */ new WeakMap();
let ee = 0, j = 1;
const Ae = 30;
let b;
const B = Symbol(process.env.NODE_ENV !== "production" ? "iterate" : ""), ze = Symbol(process.env.NODE_ENV !== "production" ? "Map key iterate" : "");
class yt {
  constructor(t, n = null, s) {
    this.fn = t, this.scheduler = n, this.active = !0, this.deps = [], this.parent = void 0, Sn(this, s);
  }
  run() {
    if (!this.active)
      return this.fn();
    let t = b, n = z;
    for (; t; ) {
      if (t === this)
        return;
      t = t.parent;
    }
    try {
      return this.parent = b, b = this, z = !0, j = 1 << ++ee, ee <= Ae ? Vn(this) : st(this), this.fn();
    } finally {
      ee <= Ae && xn(this), j = 1 << --ee, b = this.parent, z = n, this.parent = void 0, this.deferStop && this.stop();
    }
  }
  stop() {
    b === this ? this.deferStop = !0 : this.active && (st(this), this.onStop && this.onStop(), this.active = !1);
  }
}
function st(e) {
  const { deps: t } = e;
  if (t.length) {
    for (let n = 0; n < t.length; n++)
      t[n].delete(e);
    t.length = 0;
  }
}
let z = !0;
const Vt = [];
function xt() {
  Vt.push(z), z = !1;
}
function It() {
  const e = Vt.pop();
  z = e === void 0 ? !0 : e;
}
function S(e, t, n) {
  if (z && b) {
    let s = Fe.get(e);
    s || Fe.set(e, s = /* @__PURE__ */ new Map());
    let r = s.get(n);
    r || s.set(n, r = oe());
    const o = process.env.NODE_ENV !== "production" ? { effect: b, target: e, type: t, key: n } : void 0;
    je(r, o);
  }
}
function je(e, t) {
  let n = !1;
  ee <= Ae ? St(e) || (e.n |= j, n = !Ot(e)) : n = !e.has(b), n && (e.add(b), b.deps.push(e), process.env.NODE_ENV !== "production" && b.onTrack && b.onTrack(
    x(
      {
        effect: b
      },
      t
    )
  ));
}
function K(e, t, n, s, r, o) {
  const i = Fe.get(e);
  if (!i)
    return;
  let c = [];
  if (t === "clear")
    c = [...i.values()];
  else if (n === "length" && h(e)) {
    const a = Number(s);
    i.forEach((f, l) => {
      (l === "length" || l >= a) && c.push(f);
    });
  } else
    switch (n !== void 0 && c.push(i.get(n)), t) {
      case "add":
        h(e) ? Je(n) && c.push(i.get("length")) : (c.push(i.get(B)), Q(e) && c.push(i.get(ze)));
        break;
      case "delete":
        h(e) || (c.push(i.get(B)), Q(e) && c.push(i.get(ze)));
        break;
      case "set":
        Q(e) && c.push(i.get(B));
        break;
    }
  const u = process.env.NODE_ENV !== "production" ? { target: e, type: t, key: n, newValue: s, oldValue: r, oldTarget: o } : void 0;
  if (c.length === 1)
    c[0] && (process.env.NODE_ENV !== "production" ? L(c[0], u) : L(c[0]));
  else {
    const a = [];
    for (const f of c)
      f && a.push(...f);
    process.env.NODE_ENV !== "production" ? L(oe(a), u) : L(oe(a));
  }
}
function L(e, t) {
  const n = h(e) ? e : [...e];
  for (const s of n)
    s.computed && ot(s, t);
  for (const s of n)
    s.computed || ot(s, t);
}
function ot(e, t) {
  (e !== b || e.allowRecurse) && (process.env.NODE_ENV !== "production" && e.onTrigger && e.onTrigger(x({ effect: e }, t)), e.scheduler ? e.scheduler() : e.run());
}
const In = /* @__PURE__ */ cn("__proto__,__v_isRef,__isVue"), Rt = new Set(
  /* @__PURE__ */ Object.getOwnPropertyNames(Symbol).filter((e) => e !== "arguments" && e !== "caller").map((e) => Symbol[e]).filter(Ge)
), Rn = /* @__PURE__ */ qe(), Cn = /* @__PURE__ */ qe(!0), Dn = /* @__PURE__ */ qe(!0, !0), it = /* @__PURE__ */ $n();
function $n() {
  const e = {};
  return ["includes", "indexOf", "lastIndexOf"].forEach((t) => {
    e[t] = function(...n) {
      const s = p(this);
      for (let o = 0, i = this.length; o < i; o++)
        S(s, "get", o + "");
      const r = s[t](...n);
      return r === -1 || r === !1 ? s[t](...n.map(p)) : r;
    };
  }), ["push", "pop", "shift", "unshift", "splice"].forEach((t) => {
    e[t] = function(...n) {
      xt();
      const s = p(this)[t].apply(this, n);
      return It(), s;
    };
  }), e;
}
function Tn(e) {
  const t = p(this);
  return S(t, "has", e), t.hasOwnProperty(e);
}
function qe(e = !1, t = !1) {
  return function(s, r, o) {
    if (r === "__v_isReactive")
      return !e;
    if (r === "__v_isReadonly")
      return e;
    if (r === "__v_isShallow")
      return t;
    if (r === "__v_raw" && o === (e ? t ? Pt : Tt : t ? Yn : $t).get(s))
      return s;
    const i = h(s);
    if (!e) {
      if (i && m(it, r))
        return Reflect.get(it, r, o);
      if (r === "hasOwnProperty")
        return Tn;
    }
    const c = Reflect.get(s, r, o);
    return (Ge(r) ? Rt.has(r) : In(r)) || (e || S(s, "get", r), t) ? c : v(c) ? i && Je(r) ? c : c.value : O(c) ? e ? Ft(c) : Mt(c) : c;
  };
}
const Pn = /* @__PURE__ */ Mn();
function Mn(e = !1) {
  return function(n, s, r, o) {
    let i = n[s];
    if (H(i) && v(i) && !v(r))
      return !1;
    if (!e && (!Ee(r) && !H(r) && (i = p(i), r = p(r)), !h(n) && v(i) && !v(r)))
      return i.value = r, !0;
    const c = h(n) && Je(s) ? Number(s) < n.length : m(n, s), u = Reflect.set(n, s, r, o);
    return n === p(o) && (c ? se(r, i) && K(n, "set", s, r, i) : K(n, "add", s, r)), u;
  };
}
function Fn(e, t) {
  const n = m(e, t), s = e[t], r = Reflect.deleteProperty(e, t);
  return r && n && K(e, "delete", t, void 0, s), r;
}
function An(e, t) {
  const n = Reflect.has(e, t);
  return (!Ge(t) || !Rt.has(t)) && S(e, "has", t), n;
}
function zn(e) {
  return S(e, "iterate", h(e) ? "length" : B), Reflect.ownKeys(e);
}
const jn = {
  get: Rn,
  set: Pn,
  deleteProperty: Fn,
  has: An,
  ownKeys: zn
}, Ct = {
  get: Cn,
  set(e, t) {
    return process.env.NODE_ENV !== "production" && rt(
      `Set operation on key "${String(t)}" failed: target is readonly.`,
      e
    ), !0;
  },
  deleteProperty(e, t) {
    return process.env.NODE_ENV !== "production" && rt(
      `Delete operation on key "${String(t)}" failed: target is readonly.`,
      e
    ), !0;
  }
}, Kn = /* @__PURE__ */ x(
  {},
  Ct,
  {
    get: Dn
  }
), Le = (e) => e, Re = (e) => Reflect.getPrototypeOf(e);
function ue(e, t, n = !1, s = !1) {
  e = e.__v_raw;
  const r = p(e), o = p(t);
  n || (t !== o && S(r, "get", t), S(r, "get", o));
  const { has: i } = Re(r), c = s ? Le : n ? Xe : ie;
  if (i.call(r, t))
    return c(e.get(t));
  if (i.call(r, o))
    return c(e.get(o));
  e !== r && e.get(t);
}
function ae(e, t = !1) {
  const n = this.__v_raw, s = p(n), r = p(e);
  return t || (e !== r && S(s, "has", e), S(s, "has", r)), e === r ? n.has(e) : n.has(e) || n.has(r);
}
function fe(e, t = !1) {
  return e = e.__v_raw, !t && S(p(e), "iterate", B), Reflect.get(e, "size", e);
}
function ct(e) {
  e = p(e);
  const t = p(this);
  return Re(t).has.call(t, e) || (t.add(e), K(t, "add", e, e)), this;
}
function lt(e, t) {
  t = p(t);
  const n = p(this), { has: s, get: r } = Re(n);
  let o = s.call(n, e);
  o ? process.env.NODE_ENV !== "production" && Dt(n, s, e) : (e = p(e), o = s.call(n, e));
  const i = r.call(n, e);
  return n.set(e, t), o ? se(t, i) && K(n, "set", e, t, i) : K(n, "add", e, t), this;
}
function ut(e) {
  const t = p(this), { has: n, get: s } = Re(t);
  let r = n.call(t, e);
  r ? process.env.NODE_ENV !== "production" && Dt(t, n, e) : (e = p(e), r = n.call(t, e));
  const o = s ? s.call(t, e) : void 0, i = t.delete(e);
  return r && K(t, "delete", e, void 0, o), i;
}
function at() {
  const e = p(this), t = e.size !== 0, n = process.env.NODE_ENV !== "production" ? Q(e) ? new Map(e) : new Set(e) : void 0, s = e.clear();
  return t && K(e, "clear", void 0, void 0, n), s;
}
function pe(e, t) {
  return function(s, r) {
    const o = this, i = o.__v_raw, c = p(i), u = t ? Le : e ? Xe : ie;
    return !e && S(c, "iterate", B), i.forEach((a, f) => s.call(r, u(a), u(f), o));
  };
}
function de(e, t, n) {
  return function(...s) {
    const r = this.__v_raw, o = p(r), i = Q(o), c = e === "entries" || e === Symbol.iterator && i, u = e === "keys" && i, a = r[e](...s), f = n ? Le : t ? Xe : ie;
    return !t && S(
      o,
      "iterate",
      u ? ze : B
    ), {
      // iterator protocol
      next() {
        const { value: l, done: d } = a.next();
        return d ? { value: l, done: d } : {
          value: c ? [f(l[0]), f(l[1])] : f(l),
          done: d
        };
      },
      // iterable protocol
      [Symbol.iterator]() {
        return this;
      }
    };
  };
}
function M(e) {
  return function(...t) {
    if (process.env.NODE_ENV !== "production") {
      const n = t[0] ? `on key "${t[0]}" ` : "";
      console.warn(
        `${En(e)} operation ${n}failed: target is readonly.`,
        p(this)
      );
    }
    return e === "delete" ? !1 : this;
  };
}
function Hn() {
  const e = {
    get(o) {
      return ue(this, o);
    },
    get size() {
      return fe(this);
    },
    has: ae,
    add: ct,
    set: lt,
    delete: ut,
    clear: at,
    forEach: pe(!1, !1)
  }, t = {
    get(o) {
      return ue(this, o, !1, !0);
    },
    get size() {
      return fe(this);
    },
    has: ae,
    add: ct,
    set: lt,
    delete: ut,
    clear: at,
    forEach: pe(!1, !0)
  }, n = {
    get(o) {
      return ue(this, o, !0);
    },
    get size() {
      return fe(this, !0);
    },
    has(o) {
      return ae.call(this, o, !0);
    },
    add: M("add"),
    set: M("set"),
    delete: M("delete"),
    clear: M("clear"),
    forEach: pe(!0, !1)
  }, s = {
    get(o) {
      return ue(this, o, !0, !0);
    },
    get size() {
      return fe(this, !0);
    },
    has(o) {
      return ae.call(this, o, !0);
    },
    add: M("add"),
    set: M("set"),
    delete: M("delete"),
    clear: M("clear"),
    forEach: pe(!0, !0)
  };
  return ["keys", "values", "entries", Symbol.iterator].forEach((o) => {
    e[o] = de(
      o,
      !1,
      !1
    ), n[o] = de(
      o,
      !0,
      !1
    ), t[o] = de(
      o,
      !1,
      !0
    ), s[o] = de(
      o,
      !0,
      !0
    );
  }), [
    e,
    n,
    t,
    s
  ];
}
const [
  Wn,
  Bn,
  Un,
  Gn
] = /* @__PURE__ */ Hn();
function Ye(e, t) {
  const n = t ? e ? Gn : Un : e ? Bn : Wn;
  return (s, r, o) => r === "__v_isReactive" ? !e : r === "__v_isReadonly" ? e : r === "__v_raw" ? s : Reflect.get(
    m(n, r) && r in s ? n : s,
    r,
    o
  );
}
const Jn = {
  get: /* @__PURE__ */ Ye(!1, !1)
}, qn = {
  get: /* @__PURE__ */ Ye(!0, !1)
}, Ln = {
  get: /* @__PURE__ */ Ye(!0, !0)
};
function Dt(e, t, n) {
  const s = p(n);
  if (s !== n && t.call(e, s)) {
    const r = bt(e);
    console.warn(
      `Reactive ${r} contains both the raw and reactive versions of the same object${r === "Map" ? " as keys" : ""}, which can lead to inconsistencies. Avoid differentiating between the raw and reactive versions of an object and only use the reactive version if possible.`
    );
  }
}
const $t = /* @__PURE__ */ new WeakMap(), Yn = /* @__PURE__ */ new WeakMap(), Tt = /* @__PURE__ */ new WeakMap(), Pt = /* @__PURE__ */ new WeakMap();
function Qn(e) {
  switch (e) {
    case "Object":
    case "Array":
      return 1;
    case "Map":
    case "Set":
    case "WeakMap":
    case "WeakSet":
      return 2;
    default:
      return 0;
  }
}
function Xn(e) {
  return e.__v_skip || !Object.isExtensible(e) ? 0 : Qn(bt(e));
}
function Mt(e) {
  return H(e) ? e : Qe(
    e,
    !1,
    jn,
    Jn,
    $t
  );
}
function Ft(e) {
  return Qe(
    e,
    !0,
    Ct,
    qn,
    Tt
  );
}
function he(e) {
  return Qe(
    e,
    !0,
    Kn,
    Ln,
    Pt
  );
}
function Qe(e, t, n, s, r) {
  if (!O(e))
    return process.env.NODE_ENV !== "production" && console.warn(`value cannot be made reactive: ${String(e)}`), e;
  if (e.__v_raw && !(t && e.__v_isReactive))
    return e;
  const o = r.get(e);
  if (o)
    return o;
  const i = Xn(e);
  if (i === 0)
    return e;
  const c = new Proxy(
    e,
    i === 2 ? s : n
  );
  return r.set(e, c), c;
}
function U(e) {
  return H(e) ? U(e.__v_raw) : !!(e && e.__v_isReactive);
}
function H(e) {
  return !!(e && e.__v_isReadonly);
}
function Ee(e) {
  return !!(e && e.__v_isShallow);
}
function Ke(e) {
  return U(e) || H(e);
}
function p(e) {
  const t = e && e.__v_raw;
  return t ? p(t) : e;
}
function Zn(e) {
  return wn(e, "__v_skip", !0), e;
}
const ie = (e) => O(e) ? Mt(e) : e, Xe = (e) => O(e) ? Ft(e) : e;
function At(e) {
  z && b && (e = p(e), process.env.NODE_ENV !== "production" ? je(e.dep || (e.dep = oe()), {
    target: e,
    type: "get",
    key: "value"
  }) : je(e.dep || (e.dep = oe())));
}
function zt(e, t) {
  e = p(e);
  const n = e.dep;
  n && (process.env.NODE_ENV !== "production" ? L(n, {
    target: e,
    type: "set",
    key: "value",
    newValue: t
  }) : L(n));
}
function v(e) {
  return !!(e && e.__v_isRef === !0);
}
function kn(e) {
  return er(e, !1);
}
function er(e, t) {
  return v(e) ? e : new tr(e, t);
}
class tr {
  constructor(t, n) {
    this.__v_isShallow = n, this.dep = void 0, this.__v_isRef = !0, this._rawValue = n ? t : p(t), this._value = n ? t : ie(t);
  }
  get value() {
    return At(this), this._value;
  }
  set value(t) {
    const n = this.__v_isShallow || Ee(t) || H(t);
    t = n ? t : p(t), se(t, this._rawValue) && (this._rawValue = t, this._value = n ? t : ie(t), zt(this, t));
  }
}
function nr(e) {
  return v(e) ? e.value : e;
}
const rr = {
  get: (e, t, n) => nr(Reflect.get(e, t, n)),
  set: (e, t, n, s) => {
    const r = e[t];
    return v(r) && !v(n) ? (r.value = n, !0) : Reflect.set(e, t, n, s);
  }
};
function sr(e) {
  return U(e) ? e : new Proxy(e, rr);
}
class or {
  constructor(t, n, s, r) {
    this._setter = n, this.dep = void 0, this.__v_isRef = !0, this.__v_isReadonly = !1, this._dirty = !0, this.effect = new yt(t, () => {
      this._dirty || (this._dirty = !0, zt(this));
    }), this.effect.computed = this, this.effect.active = this._cacheable = !r, this.__v_isReadonly = s;
  }
  get value() {
    const t = p(this);
    return At(t), (t._dirty || !t._cacheable) && (t._dirty = !1, t._value = t.effect.run()), t._value;
  }
  set value(t) {
    this._setter(t);
  }
}
function ir(e, t, n = !1) {
  let s, r;
  const o = E(e);
  o ? (s = e, r = process.env.NODE_ENV !== "production" ? () => {
    console.warn("Write operation failed: computed value is readonly");
  } : Ue) : (s = e.get, r = e.set);
  const i = new or(s, r, o || !r, n);
  return process.env.NODE_ENV !== "production" && t && !n && (i.effect.onTrack = t.onTrack, i.effect.onTrigger = t.onTrigger), i;
}
const G = [];
function cr(e) {
  G.push(e);
}
function lr() {
  G.pop();
}
function w(e, ...t) {
  if (!process.env.NODE_ENV !== "production")
    return;
  xt();
  const n = G.length ? G[G.length - 1].component : null, s = n && n.appContext.config.warnHandler, r = ur();
  if (s)
    J(
      s,
      n,
      11,
      [
        e + t.join(""),
        n && n.proxy,
        r.map(
          ({ vnode: o }) => `at <${nn(n, o.type)}>`
        ).join(`
`),
        r
      ]
    );
  else {
    const o = [`[Vue warn]: ${e}`, ...t];
    r.length && o.push(`
`, ...ar(r)), console.warn(...o);
  }
  It();
}
function ur() {
  let e = G[G.length - 1];
  if (!e)
    return [];
  const t = [];
  for (; e; ) {
    const n = t[0];
    n && n.vnode === e ? n.recurseCount++ : t.push({
      vnode: e,
      recurseCount: 0
    });
    const s = e.component && e.component.parent;
    e = s && s.vnode;
  }
  return t;
}
function ar(e) {
  const t = [];
  return e.forEach((n, s) => {
    t.push(...s === 0 ? [] : [`
`], ...fr(n));
  }), t;
}
function fr({ vnode: e, recurseCount: t }) {
  const n = t > 0 ? `... (${t} recursive calls)` : "", s = e.component ? e.component.parent == null : !1, r = ` at <${nn(
    e.component,
    e.type,
    s
  )}`, o = ">" + n;
  return e.props ? [r, ...pr(e.props), o] : [r + o];
}
function pr(e) {
  const t = [], n = Object.keys(e);
  return n.slice(0, 3).forEach((s) => {
    t.push(...jt(s, e[s]));
  }), n.length > 3 && t.push(" ..."), t;
}
function jt(e, t, n) {
  return $(t) ? (t = JSON.stringify(t), n ? t : [`${e}=${t}`]) : typeof t == "number" || typeof t == "boolean" || t == null ? n ? t : [`${e}=${t}`] : v(t) ? (t = jt(e, p(t.value), !0), n ? t : [`${e}=Ref<`, t, ">"]) : E(t) ? [`${e}=fn${t.name ? `<${t.name}>` : ""}`] : (t = p(t), n ? t : [`${e}=`, t]);
}
const Kt = {
  sp: "serverPrefetch hook",
  bc: "beforeCreate hook",
  c: "created hook",
  bm: "beforeMount hook",
  m: "mounted hook",
  bu: "beforeUpdate hook",
  u: "updated",
  bum: "beforeUnmount hook",
  um: "unmounted hook",
  a: "activated hook",
  da: "deactivated hook",
  ec: "errorCaptured hook",
  rtc: "renderTracked hook",
  rtg: "renderTriggered hook",
  [0]: "setup function",
  [1]: "render function",
  [2]: "watcher getter",
  [3]: "watcher callback",
  [4]: "watcher cleanup function",
  [5]: "native event handler",
  [6]: "component event handler",
  [7]: "vnode hook",
  [8]: "directive hook",
  [9]: "transition hook",
  [10]: "app errorHandler",
  [11]: "app warnHandler",
  [12]: "ref function",
  [13]: "async component loader",
  [14]: "scheduler flush. This is likely a Vue internals bug. Please open an issue at https://new-issue.vuejs.org/?repo=vuejs/core"
};
function J(e, t, n, s) {
  let r;
  try {
    r = s ? e(...s) : e();
  } catch (o) {
    Ht(o, t, n);
  }
  return r;
}
function He(e, t, n, s) {
  if (E(e)) {
    const o = J(e, t, n, s);
    return o && hn(o) && o.catch((i) => {
      Ht(i, t, n);
    }), o;
  }
  const r = [];
  for (let o = 0; o < e.length; o++)
    r.push(He(e[o], t, n, s));
  return r;
}
function Ht(e, t, n, s = !0) {
  const r = t ? t.vnode : null;
  if (t) {
    let o = t.parent;
    const i = t.proxy, c = process.env.NODE_ENV !== "production" ? Kt[n] : n;
    for (; o; ) {
      const a = o.ec;
      if (a) {
        for (let f = 0; f < a.length; f++)
          if (a[f](e, i, c) === !1)
            return;
      }
      o = o.parent;
    }
    const u = t.appContext.config.errorHandler;
    if (u) {
      J(
        u,
        null,
        10,
        [e, i, c]
      );
      return;
    }
  }
  dr(e, n, r, s);
}
function dr(e, t, n, s = !0) {
  if (process.env.NODE_ENV !== "production") {
    const r = Kt[t];
    if (n && cr(n), w(`Unhandled error${r ? ` during execution of ${r}` : ""}`), n && lr(), s)
      throw e;
    console.error(e);
  } else
    console.error(e);
}
let we = !1, We = !1;
const C = [];
let A = 0;
const X = [];
let T = null, F = 0;
const Wt = /* @__PURE__ */ Promise.resolve();
let Ze = null;
const hr = 100;
function _r(e) {
  const t = Ze || Wt;
  return e ? t.then(this ? e.bind(this) : e) : t;
}
function gr(e) {
  let t = A + 1, n = C.length;
  for (; t < n; ) {
    const s = t + n >>> 1;
    ce(C[s]) < e ? t = s + 1 : n = s;
  }
  return t;
}
function ke(e) {
  (!C.length || !C.includes(
    e,
    we && e.allowRecurse ? A + 1 : A
  )) && (e.id == null ? C.push(e) : C.splice(gr(e.id), 0, e), Bt());
}
function Bt() {
  !we && !We && (We = !0, Ze = Wt.then(Gt));
}
function Ut(e) {
  h(e) ? X.push(...e) : (!T || !T.includes(
    e,
    e.allowRecurse ? F + 1 : F
  )) && X.push(e), Bt();
}
function mr(e) {
  if (X.length) {
    const t = [...new Set(X)];
    if (X.length = 0, T) {
      T.push(...t);
      return;
    }
    for (T = t, process.env.NODE_ENV !== "production" && (e = e || /* @__PURE__ */ new Map()), T.sort((n, s) => ce(n) - ce(s)), F = 0; F < T.length; F++)
      process.env.NODE_ENV !== "production" && Jt(e, T[F]) || T[F]();
    T = null, F = 0;
  }
}
const ce = (e) => e.id == null ? 1 / 0 : e.id, Er = (e, t) => {
  const n = ce(e) - ce(t);
  if (n === 0) {
    if (e.pre && !t.pre)
      return -1;
    if (t.pre && !e.pre)
      return 1;
  }
  return n;
};
function Gt(e) {
  We = !1, we = !0, process.env.NODE_ENV !== "production" && (e = e || /* @__PURE__ */ new Map()), C.sort(Er);
  const t = process.env.NODE_ENV !== "production" ? (n) => Jt(e, n) : Ue;
  try {
    for (A = 0; A < C.length; A++) {
      const n = C[A];
      if (n && n.active !== !1) {
        if (process.env.NODE_ENV !== "production" && t(n))
          continue;
        J(n, null, 14);
      }
    }
  } finally {
    A = 0, C.length = 0, mr(e), we = !1, Ze = null, (C.length || X.length) && Gt(e);
  }
}
function Jt(e, t) {
  if (!e.has(t))
    e.set(t, 1);
  else {
    const n = e.get(t);
    if (n > hr) {
      const s = t.ownerInstance, r = s && tn(s.type);
      return w(
        `Maximum recursive updates exceeded${r ? ` in component <${r}>` : ""}. This means you have a reactive effect that is mutating its own dependencies and thus recursively triggering itself. Possible sources include component template, render function, updated hook or watcher source function.`
      ), !0;
    } else
      e.set(t, n + 1);
  }
}
const k = /* @__PURE__ */ new Set();
process.env.NODE_ENV !== "production" && (Me().__VUE_HMR_RUNTIME__ = {
  createRecord: $e(wr),
  rerender: $e(Nr),
  reload: $e(br)
});
const Ne = /* @__PURE__ */ new Map();
function wr(e, t) {
  return Ne.has(e) ? !1 : (Ne.set(e, {
    initialDef: ne(t),
    instances: /* @__PURE__ */ new Set()
  }), !0);
}
function ne(e) {
  return rn(e) ? e.__vccOpts : e;
}
function Nr(e, t) {
  const n = Ne.get(e);
  n && (n.initialDef.render = t, [...n.instances].forEach((s) => {
    t && (s.render = t, ne(s.type).render = t), s.renderCache = [], s.update();
  }));
}
function br(e, t) {
  const n = Ne.get(e);
  if (!n)
    return;
  t = ne(t), ft(n.initialDef, t);
  const s = [...n.instances];
  for (const r of s) {
    const o = ne(r.type);
    k.has(o) || (o !== n.initialDef && ft(o, t), k.add(o)), r.appContext.propsCache.delete(r.type), r.appContext.emitsCache.delete(r.type), r.appContext.optionsCache.delete(r.type), r.ceReload ? (k.add(o), r.ceReload(t.styles), k.delete(o)) : r.parent ? ke(r.parent.update) : r.appContext.reload ? r.appContext.reload() : typeof window < "u" ? window.location.reload() : console.warn(
      "[HMR] Root or manually mounted instance modified. Full reload required."
    );
  }
  Ut(() => {
    for (const r of s)
      k.delete(
        ne(r.type)
      );
  });
}
function ft(e, t) {
  x(e, t);
  for (const n in e)
    n !== "__file" && !(n in t) && delete e[n];
}
function $e(e) {
  return (t, n) => {
    try {
      return e(t, n);
    } catch (s) {
      console.error(s), console.warn(
        "[HMR] Something went wrong during Vue component hot-reload. Full reload required."
      );
    }
  };
}
let R = null, vr = null;
const Or = (e) => e.__isSuspense;
function Sr(e, t) {
  t && t.pendingBranch ? h(e) ? t.effects.push(...e) : t.effects.push(e) : Ut(e);
}
const _e = {};
function yr(e, t, { immediate: n, deep: s, flush: r, onTrack: o, onTrigger: i } = P) {
  var c;
  process.env.NODE_ENV !== "production" && !t && (n !== void 0 && w(
    'watch() "immediate" option is only respected when using the watch(source, callback, options?) signature.'
  ), s !== void 0 && w(
    'watch() "deep" option is only respected when using the watch(source, callback, options?) signature.'
  ));
  const u = (g) => {
    w(
      "Invalid watch source: ",
      g,
      "A watch source can only be a getter/effect function, a ref, a reactive object, or an array of these types."
    );
  }, a = yn() === ((c = Z) == null ? void 0 : c.scope) ? Z : null;
  let f, l = !1, d = !1;
  if (v(e) ? (f = () => e.value, l = Ee(e)) : U(e) ? (f = () => e, s = !0) : h(e) ? (d = !0, l = e.some((g) => U(g) || Ee(g)), f = () => e.map((g) => {
    if (v(g))
      return g.value;
    if (U(g))
      return Y(g);
    if (E(g))
      return J(g, a, 2);
    process.env.NODE_ENV !== "production" && u(g);
  })) : E(e) ? t ? f = () => J(e, a, 2) : f = () => {
    if (!(a && a.isUnmounted))
      return _ && _(), He(
        e,
        a,
        3,
        [y]
      );
  } : (f = Ue, process.env.NODE_ENV !== "production" && u(e)), t && s) {
    const g = f;
    f = () => Y(g());
  }
  let _, y = (g) => {
    _ = I.onStop = () => {
      J(g, a, 4);
    };
  }, V = d ? new Array(e.length).fill(_e) : _e;
  const W = () => {
    if (I.active)
      if (t) {
        const g = I.run();
        (s || l || (d ? g.some(
          (sn, on) => se(sn, V[on])
        ) : se(g, V))) && (_ && _(), He(t, a, 3, [
          g,
          // pass undefined as the old value when it's changed for the first time
          V === _e ? void 0 : d && V[0] === _e ? [] : V,
          y
        ]), V = g);
      } else
        I.run();
  };
  W.allowRecurse = !!t;
  let le;
  r === "sync" ? le = W : r === "post" ? le = () => gt(W, a && a.suspense) : (W.pre = !0, a && (W.id = a.uid), le = () => ke(W));
  const I = new yt(f, le);
  return process.env.NODE_ENV !== "production" && (I.onTrack = o, I.onTrigger = i), t ? n ? W() : V = I.run() : r === "post" ? gt(
    I.run.bind(I),
    a && a.suspense
  ) : I.run(), () => {
    I.stop(), a && a.scope && fn(a.scope.effects, I);
  };
}
function Vr(e, t, n) {
  const s = this.proxy, r = $(e) ? e.includes(".") ? xr(s, e) : () => s[e] : e.bind(s, s);
  let o;
  E(t) ? o = t : (o = t.handler, n = t);
  const i = Z;
  wt(this);
  const c = yr(r, o.bind(s), n);
  return i ? wt(i) : Jr(), c;
}
function xr(e, t) {
  const n = t.split(".");
  return () => {
    let s = e;
    for (let r = 0; r < n.length && s; r++)
      s = s[n[r]];
    return s;
  };
}
function Y(e, t) {
  if (!O(e) || e.__v_skip || (t = t || /* @__PURE__ */ new Set(), t.has(e)))
    return e;
  if (t.add(e), v(e))
    Y(e.value, t);
  else if (h(e))
    for (let n = 0; n < e.length; n++)
      Y(e[n], t);
  else if (dn(e) || Q(e))
    e.forEach((n) => {
      Y(n, t);
    });
  else if (gn(e))
    for (const n in e)
      Y(e[n], t);
  return e;
}
function Ir(e, t) {
  return E(e) ? (
    // #8326: extend call and options.name access are considered side-effects
    // by Rollup, so we have to wrap it in a pure-annotated IIFE.
    /* @__PURE__ */ (() => x({ name: e.name }, t, { setup: e }))()
  ) : e;
}
const Rr = (e) => !!e.type.__asyncLoader, Cr = Symbol.for("v-ndc");
function Dr(e, t, n = {}, s, r) {
  if (R.isCE || R.parent && Rr(R.parent) && R.parent.isCE)
    return t !== "default" && (n.name = t), De("slot", n, s && s());
  let o = e[t];
  process.env.NODE_ENV !== "production" && o && o.length > 1 && (w(
    "SSR-optimized slot function detected in a non-SSR-optimized render function. You need to mark this component with $dynamic-slots in the parent template."
  ), o = () => []), o && o._c && (o._d = !1), Oe();
  const i = o && qt(o(n)), c = Yt(
    Ce,
    {
      key: n.key || // slot content array of a dynamic conditional slot may have a branch
      // key attached in the `createSlots` helper, respect that
      i && i.key || `_${t}`
    },
    i || (s ? s() : []),
    i && e._ === 1 ? 64 : -2
  );
  return !r && c.scopeId && (c.slotScopeIds = [c.scopeId + "-s"]), o && o._c && (o._d = !0), c;
}
function qt(e) {
  return e.some((t) => Qt(t) ? !(t.type === ve || t.type === Ce && !qt(t.children)) : !0) ? e : null;
}
const Be = (e) => e ? qr(e) ? Yr(e) || e.proxy : Be(e.parent) : null, re = (
  // Move PURE marker to new line to workaround compiler discarding it
  // due to type annotation
  /* @__PURE__ */ x(/* @__PURE__ */ Object.create(null), {
    $: (e) => e,
    $el: (e) => e.vnode.el,
    $data: (e) => e.data,
    $props: (e) => process.env.NODE_ENV !== "production" ? he(e.props) : e.props,
    $attrs: (e) => process.env.NODE_ENV !== "production" ? he(e.attrs) : e.attrs,
    $slots: (e) => process.env.NODE_ENV !== "production" ? he(e.slots) : e.slots,
    $refs: (e) => process.env.NODE_ENV !== "production" ? he(e.refs) : e.refs,
    $parent: (e) => Be(e.parent),
    $root: (e) => Be(e.root),
    $emit: (e) => e.emit,
    $options: (e) => Pr(e),
    $forceUpdate: (e) => e.f || (e.f = () => ke(e.update)),
    $nextTick: (e) => e.n || (e.n = _r.bind(e.proxy)),
    $watch: (e) => Vr.bind(e)
  })
), $r = (e) => e === "_" || e === "$", Te = (e, t) => e !== P && !e.__isScriptSetup && m(e, t), Tr = {
  get({ _: e }, t) {
    const { ctx: n, setupState: s, data: r, props: o, accessCache: i, type: c, appContext: u } = e;
    if (process.env.NODE_ENV !== "production" && t === "__isVue")
      return !0;
    let a;
    if (t[0] !== "$") {
      const _ = i[t];
      if (_ !== void 0)
        switch (_) {
          case 1:
            return s[t];
          case 2:
            return r[t];
          case 4:
            return n[t];
          case 3:
            return o[t];
        }
      else {
        if (Te(s, t))
          return i[t] = 1, s[t];
        if (r !== P && m(r, t))
          return i[t] = 2, r[t];
        if (
          // only cache other properties when instance has declared (thus stable)
          // props
          (a = e.propsOptions[0]) && m(a, t)
        )
          return i[t] = 3, o[t];
        if (n !== P && m(n, t))
          return i[t] = 4, n[t];
        i[t] = 0;
      }
    }
    const f = re[t];
    let l, d;
    if (f)
      return t === "$attrs" ? (S(e, "get", t), process.env.NODE_ENV !== "production" && void 0) : process.env.NODE_ENV !== "production" && t === "$slots" && S(e, "get", t), f(e);
    if (
      // css module (injected by vue-loader)
      (l = c.__cssModules) && (l = l[t])
    )
      return l;
    if (n !== P && m(n, t))
      return i[t] = 4, n[t];
    if (
      // global properties
      d = u.config.globalProperties, m(d, t)
    )
      return d[t];
    process.env.NODE_ENV !== "production" && R && (!$(t) || // #1091 avoid internal isRef/isVNode checks on component instance leading
    // to infinite warning loop
    t.indexOf("__v") !== 0) && (r !== P && $r(t[0]) && m(r, t) ? w(
      `Property ${JSON.stringify(
        t
      )} must be accessed via $data because it starts with a reserved character ("$" or "_") and is not proxied on the render context.`
    ) : e === R && w(
      `Property ${JSON.stringify(t)} was accessed during render but is not defined on instance.`
    ));
  },
  set({ _: e }, t, n) {
    const { data: s, setupState: r, ctx: o } = e;
    return Te(r, t) ? (r[t] = n, !0) : process.env.NODE_ENV !== "production" && r.__isScriptSetup && m(r, t) ? (w(`Cannot mutate <script setup> binding "${t}" from Options API.`), !1) : s !== P && m(s, t) ? (s[t] = n, !0) : m(e.props, t) ? (process.env.NODE_ENV !== "production" && w(`Attempting to mutate prop "${t}". Props are readonly.`), !1) : t[0] === "$" && t.slice(1) in e ? (process.env.NODE_ENV !== "production" && w(
      `Attempting to mutate public property "${t}". Properties starting with $ are reserved and readonly.`
    ), !1) : (process.env.NODE_ENV !== "production" && t in e.appContext.config.globalProperties ? Object.defineProperty(o, t, {
      enumerable: !0,
      configurable: !0,
      value: n
    }) : o[t] = n, !0);
  },
  has({
    _: { data: e, setupState: t, accessCache: n, ctx: s, appContext: r, propsOptions: o }
  }, i) {
    let c;
    return !!n[i] || e !== P && m(e, i) || Te(t, i) || (c = o[0]) && m(c, i) || m(s, i) || m(re, i) || m(r.config.globalProperties, i);
  },
  defineProperty(e, t, n) {
    return n.get != null ? e._.accessCache[t] = 0 : m(n, "value") && this.set(e, t, n.value, null), Reflect.defineProperty(e, t, n);
  }
};
process.env.NODE_ENV !== "production" && (Tr.ownKeys = (e) => (w(
  "Avoid app logic that relies on enumerating keys on a component instance. The keys will be empty in production mode to avoid performance overhead."
), Reflect.ownKeys(e)));
function pt(e) {
  return h(e) ? e.reduce(
    (t, n) => (t[n] = null, t),
    {}
  ) : e;
}
function Pr(e) {
  const t = e.type, { mixins: n, extends: s } = t, {
    mixins: r,
    optionsCache: o,
    config: { optionMergeStrategies: i }
  } = e.appContext, c = o.get(t);
  let u;
  return c ? u = c : !r.length && !n && !s ? u = t : (u = {}, r.length && r.forEach(
    (a) => be(u, a, i, !0)
  ), be(u, t, i)), O(t) && o.set(t, u), u;
}
function be(e, t, n, s = !1) {
  const { mixins: r, extends: o } = t;
  o && be(e, o, n, !0), r && r.forEach(
    (i) => be(e, i, n, !0)
  );
  for (const i in t)
    if (s && i === "expose")
      process.env.NODE_ENV !== "production" && w(
        '"expose" option is ignored when declared in mixins or extends. It should only be declared in the base component itself.'
      );
    else {
      const c = Mr[i] || n && n[i];
      e[i] = c ? c(e[i], t[i]) : t[i];
    }
  return e;
}
const Mr = {
  data: dt,
  props: _t,
  emits: _t,
  // objects
  methods: te,
  computed: te,
  // lifecycle
  beforeCreate: N,
  created: N,
  beforeMount: N,
  mounted: N,
  beforeUpdate: N,
  updated: N,
  beforeDestroy: N,
  beforeUnmount: N,
  destroyed: N,
  unmounted: N,
  activated: N,
  deactivated: N,
  errorCaptured: N,
  serverPrefetch: N,
  // assets
  components: te,
  directives: te,
  // watch
  watch: Ar,
  // provide / inject
  provide: dt,
  inject: Fr
};
function dt(e, t) {
  return t ? e ? function() {
    return x(
      E(e) ? e.call(this, this) : e,
      E(t) ? t.call(this, this) : t
    );
  } : t : e;
}
function Fr(e, t) {
  return te(ht(e), ht(t));
}
function ht(e) {
  if (h(e)) {
    const t = {};
    for (let n = 0; n < e.length; n++)
      t[e[n]] = e[n];
    return t;
  }
  return e;
}
function N(e, t) {
  return e ? [...new Set([].concat(e, t))] : t;
}
function te(e, t) {
  return e ? x(/* @__PURE__ */ Object.create(null), e, t) : t;
}
function _t(e, t) {
  return e ? h(e) && h(t) ? [.../* @__PURE__ */ new Set([...e, ...t])] : x(
    /* @__PURE__ */ Object.create(null),
    pt(e),
    pt(t ?? {})
  ) : t;
}
function Ar(e, t) {
  if (!e)
    return t;
  if (!t)
    return e;
  const n = x(/* @__PURE__ */ Object.create(null), e);
  for (const s in t)
    n[s] = N(e[s], t[s]);
  return n;
}
const gt = Sr, zr = (e) => e.__isTeleport, Ce = Symbol.for("v-fgt"), jr = Symbol.for("v-txt"), ve = Symbol.for("v-cmt"), ge = [];
let D = null;
function Oe(e = !1) {
  ge.push(D = e ? null : []);
}
function Kr() {
  ge.pop(), D = ge[ge.length - 1] || null;
}
function Lt(e) {
  return e.dynamicChildren = D || ln, Kr(), D && D.push(e), e;
}
function mt(e, t, n, s, r, o) {
  return Lt(
    Se(
      e,
      t,
      n,
      s,
      r,
      o,
      !0
      /* isBlock */
    )
  );
}
function Yt(e, t, n, s, r) {
  return Lt(
    De(
      e,
      t,
      n,
      s,
      r,
      !0
      /* isBlock: prevent a block from tracking itself */
    )
  );
}
function Qt(e) {
  return e ? e.__v_isVNode === !0 : !1;
}
const Hr = (...e) => kt(
  ...e
), Xt = "__vInternal", Zt = ({ key: e }) => e ?? null, me = ({
  ref: e,
  ref_key: t,
  ref_for: n
}) => (typeof e == "number" && (e = "" + e), e != null ? $(e) || v(e) || E(e) ? { i: R, r: e, k: t, f: !!n } : e : null);
function Se(e, t = null, n = null, s = 0, r = null, o = e === Ce ? 0 : 1, i = !1, c = !1) {
  const u = {
    __v_isVNode: !0,
    __v_skip: !0,
    type: e,
    props: t,
    key: t && Zt(t),
    ref: t && me(t),
    scopeId: vr,
    slotScopeIds: null,
    children: n,
    component: null,
    suspense: null,
    ssContent: null,
    ssFallback: null,
    dirs: null,
    transition: null,
    el: null,
    anchor: null,
    target: null,
    targetAnchor: null,
    staticCount: 0,
    shapeFlag: o,
    patchFlag: s,
    dynamicProps: r,
    dynamicChildren: null,
    appContext: null,
    ctx: R
  };
  return c ? (et(u, n), o & 128 && e.normalize(u)) : n && (u.shapeFlag |= $(n) ? 8 : 16), process.env.NODE_ENV !== "production" && u.key !== u.key && w("VNode created with invalid key (NaN). VNode type:", u.type), // avoid a block node from tracking itself
  !i && // has current parent block
  D && // presence of a patch flag indicates this node needs patching on updates.
  // component nodes also should always be patched, because even if the
  // component doesn't need to update, it needs to persist the instance on to
  // the next vnode so that it can be properly unmounted later.
  (u.patchFlag > 0 || o & 6) && // the EVENTS flag is only for hydration and if it is the only flag, the
  // vnode should not be considered dynamic due to handler caching.
  u.patchFlag !== 32 && D.push(u), u;
}
const De = process.env.NODE_ENV !== "production" ? Hr : kt;
function kt(e, t = null, n = null, s = 0, r = null, o = !1) {
  if ((!e || e === Cr) && (process.env.NODE_ENV !== "production" && !e && w(`Invalid vnode type when creating vnode: ${e}.`), e = ve), Qt(e)) {
    const c = ye(
      e,
      t,
      !0
      /* mergeRef: true */
    );
    return n && et(c, n), !o && D && (c.shapeFlag & 6 ? D[D.indexOf(e)] = c : D.push(c)), c.patchFlag |= -2, c;
  }
  if (rn(e) && (e = e.__vccOpts), t) {
    t = Wr(t);
    let { class: c, style: u } = t;
    c && !$(c) && (t.class = Ie(c)), O(u) && (Ke(u) && !h(u) && (u = x({}, u)), t.style = xe(u));
  }
  const i = $(e) ? 1 : Or(e) ? 128 : zr(e) ? 64 : O(e) ? 4 : E(e) ? 2 : 0;
  return process.env.NODE_ENV !== "production" && i & 4 && Ke(e) && (e = p(e), w(
    "Vue received a Component which was made a reactive object. This can lead to unnecessary performance overhead, and should be avoided by marking the component with `markRaw` or using `shallowRef` instead of `ref`.",
    `
Component that was made reactive: `,
    e
  )), Se(
    e,
    t,
    n,
    s,
    r,
    i,
    o,
    !0
  );
}
function Wr(e) {
  return e ? Ke(e) || Xt in e ? x({}, e) : e : null;
}
function ye(e, t, n = !1) {
  const { props: s, ref: r, patchFlag: o, children: i } = e, c = t ? Gr(s || {}, t) : s;
  return {
    __v_isVNode: !0,
    __v_skip: !0,
    type: e.type,
    props: c,
    key: c && Zt(c),
    ref: t && t.ref ? (
      // #2078 in the case of <component :is="vnode" ref="extra"/>
      // if the vnode itself already has a ref, cloneVNode will need to merge
      // the refs so the single vnode can be set on multiple refs
      n && r ? h(r) ? r.concat(me(t)) : [r, me(t)] : me(t)
    ) : r,
    scopeId: e.scopeId,
    slotScopeIds: e.slotScopeIds,
    children: process.env.NODE_ENV !== "production" && o === -1 && h(i) ? i.map(en) : i,
    target: e.target,
    targetAnchor: e.targetAnchor,
    staticCount: e.staticCount,
    shapeFlag: e.shapeFlag,
    // if the vnode is cloned with extra props, we can no longer assume its
    // existing patch flag to be reliable and need to add the FULL_PROPS flag.
    // note: preserve flag for fragments since they use the flag for children
    // fast paths only.
    patchFlag: t && e.type !== Ce ? o === -1 ? 16 : o | 16 : o,
    dynamicProps: e.dynamicProps,
    dynamicChildren: e.dynamicChildren,
    appContext: e.appContext,
    dirs: e.dirs,
    transition: e.transition,
    // These should technically only be non-null on mounted VNodes. However,
    // they *should* be copied for kept-alive vnodes. So we just always copy
    // them since them being non-null during a mount doesn't affect the logic as
    // they will simply be overwritten.
    component: e.component,
    suspense: e.suspense,
    ssContent: e.ssContent && ye(e.ssContent),
    ssFallback: e.ssFallback && ye(e.ssFallback),
    el: e.el,
    anchor: e.anchor,
    ctx: e.ctx,
    ce: e.ce
  };
}
function en(e) {
  const t = ye(e);
  return h(e.children) && (t.children = e.children.map(en)), t;
}
function Br(e = " ", t = 0) {
  return De(jr, null, e, t);
}
function Ur(e = "", t = !1) {
  return t ? (Oe(), Yt(ve, null, e)) : De(ve, null, e);
}
function et(e, t) {
  let n = 0;
  const { shapeFlag: s } = e;
  if (t == null)
    t = null;
  else if (h(t))
    n = 16;
  else if (typeof t == "object")
    if (s & 65) {
      const r = t.default;
      r && (r._c && (r._d = !1), et(e, r()), r._c && (r._d = !0));
      return;
    } else {
      n = 32;
      const r = t._;
      !r && !(Xt in t) ? t._ctx = R : r === 3 && R && (R.slots._ === 1 ? t._ = 1 : (t._ = 2, e.patchFlag |= 1024));
    }
  else
    E(t) ? (t = { default: t, _ctx: R }, n = 32) : (t = String(t), s & 64 ? (n = 16, t = [Br(t)]) : n = 8);
  e.children = t, e.shapeFlag |= n;
}
function Gr(...e) {
  const t = {};
  for (let n = 0; n < e.length; n++) {
    const s = e[n];
    for (const r in s)
      if (r === "class")
        t.class !== s.class && (t.class = Ie([t.class, s.class]));
      else if (r === "style")
        t.style = xe([t.style, s.style]);
      else if (an(r)) {
        const o = t[r], i = s[r];
        i && o !== i && !(h(o) && o.includes(i)) && (t[r] = o ? [].concat(o, i) : i);
      } else
        r !== "" && (t[r] = s[r]);
  }
  return t;
}
let Z = null, tt, q, Et = "__VUE_INSTANCE_SETTERS__";
(q = Me()[Et]) || (q = Me()[Et] = []), q.push((e) => Z = e), tt = (e) => {
  q.length > 1 ? q.forEach((t) => t(e)) : q[0](e);
};
const wt = (e) => {
  tt(e), e.scope.on();
}, Jr = () => {
  Z && Z.scope.off(), tt(null);
};
function qr(e) {
  return e.vnode.shapeFlag & 4;
}
let Lr = !1;
function Yr(e) {
  if (e.exposed)
    return e.exposeProxy || (e.exposeProxy = new Proxy(sr(Zn(e.exposed)), {
      get(t, n) {
        if (n in t)
          return t[n];
        if (n in re)
          return re[n](e);
      },
      has(t, n) {
        return n in t || n in re;
      }
    }));
}
const Qr = /(?:^|[-_])(\w)/g, Xr = (e) => e.replace(Qr, (t) => t.toUpperCase()).replace(/[-_]/g, "");
function tn(e, t = !0) {
  return E(e) ? e.displayName || e.name : e.name || t && e.__name;
}
function nn(e, t, n = !1) {
  let s = tn(t);
  if (!s && t.__file) {
    const r = t.__file.match(/([^/\\]+)\.\w+$/);
    r && (s = r[1]);
  }
  if (!s && e && e.parent) {
    const r = (o) => {
      for (const i in o)
        if (o[i] === t)
          return i;
    };
    s = r(
      e.components || e.parent.type.components
    ) || r(e.appContext.components);
  }
  return s ? Xr(s) : n ? "App" : "Anonymous";
}
function rn(e) {
  return E(e) && "__vccOpts" in e;
}
const Zr = (e, t) => ir(e, t, Lr);
function Pe(e) {
  return !!(e && e.__v_isShallow);
}
function kr() {
  if (!process.env.NODE_ENV !== "production" || typeof window > "u")
    return;
  const e = { style: "color:#3ba776" }, t = { style: "color:#0b1bc9" }, n = { style: "color:#b62e24" }, s = { style: "color:#9d288c" }, r = {
    header(l) {
      return O(l) ? l.__isVue ? ["div", e, "VueInstance"] : v(l) ? [
        "div",
        {},
        ["span", e, f(l)],
        "<",
        c(l.value),
        ">"
      ] : U(l) ? [
        "div",
        {},
        ["span", e, Pe(l) ? "ShallowReactive" : "Reactive"],
        "<",
        c(l),
        `>${H(l) ? " (readonly)" : ""}`
      ] : H(l) ? [
        "div",
        {},
        ["span", e, Pe(l) ? "ShallowReadonly" : "Readonly"],
        "<",
        c(l),
        ">"
      ] : null : null;
    },
    hasBody(l) {
      return l && l.__isVue;
    },
    body(l) {
      if (l && l.__isVue)
        return [
          "div",
          {},
          ...o(l.$)
        ];
    }
  };
  function o(l) {
    const d = [];
    l.type.props && l.props && d.push(i("props", p(l.props))), l.setupState !== P && d.push(i("setup", l.setupState)), l.data !== P && d.push(i("data", p(l.data)));
    const _ = u(l, "computed");
    _ && d.push(i("computed", _));
    const y = u(l, "inject");
    return y && d.push(i("injected", y)), d.push([
      "div",
      {},
      [
        "span",
        {
          style: s.style + ";opacity:0.66"
        },
        "$ (internal): "
      ],
      ["object", { object: l }]
    ]), d;
  }
  function i(l, d) {
    return d = x({}, d), Object.keys(d).length ? [
      "div",
      { style: "line-height:1.25em;margin-bottom:0.6em" },
      [
        "div",
        {
          style: "color:#476582"
        },
        l
      ],
      [
        "div",
        {
          style: "padding-left:1.25em"
        },
        ...Object.keys(d).map((_) => [
          "div",
          {},
          ["span", s, _ + ": "],
          c(d[_], !1)
        ])
      ]
    ] : ["span", {}];
  }
  function c(l, d = !0) {
    return typeof l == "number" ? ["span", t, l] : typeof l == "string" ? ["span", n, JSON.stringify(l)] : typeof l == "boolean" ? ["span", s, l] : O(l) ? ["object", { object: d ? p(l) : l }] : ["span", n, String(l)];
  }
  function u(l, d) {
    const _ = l.type;
    if (E(_))
      return;
    const y = {};
    for (const V in l.ctx)
      a(_, V, d) && (y[V] = l.ctx[V]);
    return y;
  }
  function a(l, d, _) {
    const y = l[_];
    if (h(y) && y.includes(d) || O(y) && d in y || l.extends && a(l.extends, d, _) || l.mixins && l.mixins.some((V) => a(V, d, _)))
      return !0;
  }
  function f(l) {
    return Pe(l) ? "ShallowRef" : l.effect ? "ComputedRef" : "Ref";
  }
  window.devtoolsFormatters ? window.devtoolsFormatters.push(r) : window.devtoolsFormatters = [r];
}
function es() {
  kr();
}
process.env.NODE_ENV !== "production" && es();
const ts = ["type", "value", "disabled", "placeholder", "autofocus"], ns = /* @__PURE__ */ Ir({
  name: "PInput",
  __name: "index",
  props: {
    modelValue: {
      type: [String, Number],
      default: ""
    },
    type: {
      type: String,
      default: "text"
    },
    size: {
      type: String,
      default: "default"
    },
    disabled: Boolean,
    clearable: Boolean,
    placeholder: {
      type: String,
      default: "请输入内容"
    },
    //当页面加载时 <input> 元素自动获得焦点
    autofocus: Boolean
  },
  emits: ["update:modelValue", "clear", "focus", "blur", "input", "change"],
  setup(e, { emit: t }) {
    const n = e;
    console.log(n);
    const s = kn(!1), r = (f) => {
      let { value: l } = f.target;
      console.log(l, "valuevalue"), t("update:modelValue", l), t("input", l);
    }, o = () => {
      t("update:modelValue", ""), t("change", ""), t("clear"), t("input", "");
    }, i = (f) => {
      s.value = !0, t("focus", f);
    }, c = (f) => {
      s.value = !1, t("blur", f);
    }, u = (f) => {
      t("change", f.target.value);
    }, a = Zr(() => [
      n.clearable ? "p-input-clearable" : n.size == "default" ? "p-input-default" : `p-input-${n.size}`,
      n.disabled ? "p-input-disabled" : ""
    ]);
    return (f, l) => (Oe(), mt("div", {
      class: "p-input-default",
      style: xe({ width: e.size && e.size !== "" ? `${e.size}px` : "" })
    }, [
      Se("div", {
        class: Ie(a.value)
      }, [
        Se("input", {
          ref: "input",
          type: e.type,
          value: e.modelValue,
          disabled: e.disabled,
          placeholder: e.placeholder,
          autofocus: e.autofocus,
          onInput: r,
          onFocus: i,
          onBlur: c,
          onChange: u
        }, null, 40, ts),
        e.clearable && e.modelValue != "" ? (Oe(), mt("i", {
          key: 0,
          class: "clearable-icon",
          onClick: o
        }, "x")) : Ur("", !0)
      ], 2),
      Dr(f.$slots, "btn", {}, void 0, !0)
    ], 4));
  }
});
const rs = (e, t) => {
  const n = e.__vccOpts || e;
  for (const [s, r] of t)
    n[s] = r;
  return n;
}, Nt = /* @__PURE__ */ rs(ns, [["__scopeId", "data-v-8ec2f7fc"]]), ss = {
  install: (e) => {
    e.component(Nt.name, Nt);
  }
}, os = [ss], is = (e) => {
  os.forEach((t) => {
    e.use(t);
  });
}, ls = {
  version: "0.0.1",
  install: is
};
export {
  ss as PInput,
  ls as default,
  is as install
};
